### DISCUSSION TITLE
how ican make blueman pair any device automaticliy without message box

### DISCUSSION THREAD

**Discussion by wesamdev** [blueman-project/blueman#2299] [Q&A]
please help tomowrrrow is test of the project in school i have 5h only

**Comment by cschramm** [comment]
Not sure how that question is meant. If you generally want to auto-pair devices, blueman is not the right tool and you might just want to write you own agent. If you want to hack blueman accordingly, you'll want to look at `blueman/main/applet/BluezAgent.py`, probably mainly `_on_request_confirmation`.

**Reply by wesamdev** [reply]
how i can write one

**Reply by cschramm** [reply]
By implementing the `org.bluez.Agent` D-Bus interface (`man org.bluez.Agent`) and registering the object with the `org.bluez.AgentManager` (`man org.bluez.AgentManager`).

**Reply by wesamdev** [reply]
idk how this works did you have easy docs

**Reply by wesamdev** [reply]
i get this eror when i try to compile blueman:
```make[3]: Entering directory '/home/wesam/blueman/po'
rm -f af.gmo && : -c --statistics --verbose -o af.gmo af.po
mv: cannot stat 't-af.gmo': No such file or directory
make[3]: *** [Makefile:289: af.gmo] Error 1
make[3]: Leaving directory '/home/wesam/blueman/po'
make[2]: *** [Makefile:327: stamp-po] Error 2
make[2]: Leaving directory '/home/wesam/blueman/po'
make[1]: *** [Makefile:561: all-recursive] Error 1
make[1]: Leaving directory '/home/wesa```

**Reply by cschramm** [reply]
Those are BlueZ interfaces, documented in their respective man pages and you can find a bazillion of D-Bus tutorials for whatever tech stack you want to use on the Internets. It has nothing to do with blueman.

**Reply by wesamdev** [reply]
i found solution by run this python code :
 ```python
 from bluez_peripheral.util import Adapter, get_message_bus
from bluez_peripheral.advert import Advertisement
from bluez_peripheral.agent import NoIoAgent
import asyncio

async def main():
    # Alternativly you can request this bus directly from dbus_next.
    bus = await get_message_bus()

    # An agent is required to handle pairing 
    agent = NoIoAgent()
    # This script needs superuser for this to work.
    await agent.register(bus)


    await bus.wait_for_disconnect()

if __name__ == "__main__":
    asyncio.run(main())

```

and dont forgot to install bluez_peripheral:
```sh
pip install bluez_peripheral
```
